test/e2e/velero_utils.go

```diff
@@ -192,24 +200,86 @@ func CheckRestorePhase(ctx context.Context, veleroCLI string, restoreName string
+	err = CheckBackupPhase(ctx, veleroCLI, veleroNamespace, backupName, velerov1api.BackupPhaseCompleted)
+
+	return err
 }
 
 // VeleroRestore uses the veleroCLI to restore from a Velero backup.
-func VeleroRestore(ctx context.Context, veleroCLI string, restoreName string, backupName string) error {
-	restoreCmd := exec.CommandContext(ctx, veleroCLI, "create", "restore", restoreName, "--from-backup", backupName, "--wait")
+func VeleroRestore(ctx context.Context, veleroCLI string, veleroNamespace string, restoreName string, backupName string) error {
+	restoreCmd := exec.CommandContext(ctx, veleroCLI, "--namespace", veleroNamespace, "create", "restore", restoreName,
+		"--from-backup", backupName, "--wait")
+
+	restoreCmd.Stdout = os.Stdout
+	restoreCmd.Stderr = os.Stderr
 	fmt.Printf("restore cmd =%v\n", restoreCmd)
 	err := restoreCmd.Run()
 	if err != nil {
 		return err
 	}
-	return CheckRestorePhase(ctx, veleroCLI, restoreName, velerov1api.RestorePhaseCompleted)
+	return CheckRestorePhase(ctx, veleroCLI, veleroNamespace, restoreName, velerov1api.RestorePhaseCompleted)
+}
+
+func VeleroInstall(ctx context.Context, veleroNamespace string, cloudProvider string, objectStoreProvider string, useVolumeSnapshots bool,
+	cloudCredentialsFile string, bslBucket string, bslPrefix string, bslConfig string, vslConfig string) error {
+
+	if cloudProvider != "kind" {
+		if objectStoreProvider != "" {
+			return errors.New("For cloud platforms, object store plugin cannot be overridden") // Can't set an object store provider that is different than your cloud
+		}
+		objectStoreProvider = cloudProvider
+	} else {
+		if objectStoreProvider == "" {
+			return errors.New("No object store provider specified - must be specified when using kind as the cloud provider") // Gotta have an object store provider
+		}
+	}
+	err := EnsureClusterExists(ctx)
+	if err != nil {
+		return errors.WithMessage(err, "Failed to ensure kubernetes cluster exists")
+	}
+	veleroInstallOptions, err := GetProviderVeleroInstallOptions(objectStoreProvider, cloudCredentialsFile, bslBucket, bslPrefix, bslConfig, vslConfig, getProviderPlugins(objectStoreProvider))
+	if err != nil {
+		return errors.WithMessagef(err, "Failed to get Velero InstallOptions for plugin provider %s", objectStoreProvider)
+	}
+	veleroInstallOptions.UseRestic = !useVolumeSnapshots
+	veleroInstallOptions.Namespace = veleroNamespace
+	err = InstallVeleroServer(veleroInstallOptions)
+	if err != nil {
+		return errors.WithMessagef(err, "Failed to install Velero in cluster")
+	}
+	return nil
+}
+
+func VeleroUninstall(ctx context.Context, client *kubernetes.Clientset, veleroNamespace string) error {
+	return client.CoreV1().Namespaces().Delete(ctx, veleroNamespace, metav1.DeleteOptions{})
+}
+
+func VeleroBackupLogs(ctx context.Context, veleroCLI string, veleroNamespace string, backupName string) error {
+	describeCmd := exec.CommandContext(ctx, veleroCLI, "--namespace", veleroNamespace, "backup", "describe", backupName)
```

Comment:
Could we set the velero namespace with `velero client config set namepace veleroNamepace` instead of passing it around all the time?

Reply:
We could, but this also allows us to test multiple veleros at the same time.  Let's revisit that later.